buildtest/menu/build.py

```diff
@@ -10,58 +10,58 @@
 
 from buildtest.defaults import TESTCONFIG_ROOT, BUILDTEST_CONFIG_FILE
 
-from buildtest.buildsystem.base import BuildConfig
+from buildtest.buildsystem.base import BuildspecParser
 from buildtest.config import load_configuration, check_configuration
 from buildtest.executors.base import BuildExecutor
-from buildtest.utils.file import walk_tree, resolve_path
+from buildtest.utils.file import walk_tree, resolve_path, is_dir, is_file
 
 logger = logging.getLogger(__name__)
 
 
-def discover_configs(config_file):
-    """Given a config file specified by the user with buildtest build -c,
+def discover_buildspecs(buildspec):
+    """Given a buildspec file specified by the user with ``buildtest build --buildspec``,
        discover one or more files and return a list for buildtest to parse.
        Examples of intended functionality are documented here. For all of
        the below, test config root refers to $HOME/.buildtest/site
  
        # A relative path to a file in the PWD (outside of test config root, returns single)
-       buildtest build -c relative-folder/hello.sh.yml
+       buildtest build --buildspec relative-folder/hello.sh.yml
 
        # A relative path to a file in build test root (returns single)
-       buildtest build -c buildtest build -c github.com/HPC-buildtest/tutorials/hello-world/hello.sh.ym
+       buildtest build --buildspec github.com/HPC-buildtest/tutorials/hello-world/hello.sh.yml
 
        # relative directory path (returns multiple)
-       buildtest build -c hello-world
+       buildtest build --buildspec hello-world
 
        # relative directory path in build test root (returns multiple)
-       buildtest build -c github.com/HPC-buildtest/tutorials/hello-world/
+       buildtest build --buildspec github.com/HPC-buildtest/tutorials/hello-world/
     """
 
-    config_files = []
+    buildspec_files = []
 
     # If no config file provided, assume discovering across buildtest/site
-    if not config_file:
+    if not buildspec:
```

Comment:
I think this config_file needs to be changed?

Reply:
yep agreed good catch. I renamed this as ``settings_file``.

Reply:
Wait, so buildspec == settings_file ? I'm confused. If this is the settings_file, then why is it called buildspec?